src/transcoding.ts

```diff
@@ -237,27 +237,45 @@ export function flattenObject(request: JSONObject): JSONObject {
 
 export function requestChangeCaseAndCleanup(
   request: JSONObject,
-  caseChangeFunc: (key: string) => string
+  caseChangeFunc: (key: string) => string,
+  fieldsToChange?: string[]
 ) {
   if (!request || typeof request !== 'object') {
     return request;
   }
+  const fieldsToChangeWithFunc = fieldsToChange?.map(x => caseChangeFunc(x));
   const convertedRequest: JSONObject = {};
   for (const field in request) {
     // cleaning up inherited properties
     if (!Object.prototype.hasOwnProperty.call(request, field)) {
       continue;
     }
-    const convertedField = caseChangeFunc(field);
+    let convertedField = caseChangeFunc(field);
+
+    // Here, we want to check if the fields in the proto match
+    // the fields we are changing; if not, we assume it's user
+    // input and revert back to its original form
+    if (
+      fieldsToChangeWithFunc &&
+      !fieldsToChangeWithFunc?.includes(convertedField)
```

Comment:
This is going to take O(N²): for each field to check, `includes()` will iterate the whole `fieldsToChangeWithFunc`. If you change `fieldsToChangeWithFunc` to a `Set<string>`, the `.has()` check will be faster.

Reply:
cool!